api/types/databaseobject/databaseobject.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package databaseobject
+
+import (
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/defaults"
+	dbobjectv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobject/v1"
+	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
+	"github.com/gravitational/teleport/api/types"
+)
+
+// NewDatabaseObject creates a new dbobjectv1.DatabaseObject.
+func NewDatabaseObject(name string, spec *dbobjectv1.DatabaseObjectSpec) (*dbobjectv1.DatabaseObject, error) {
+	return NewDatabaseObjectWithLabels(name, nil, spec)
+}
+
+// NewDatabaseObjectWithLabels creates a new dbobjectv1.DatabaseObject with specified labels.
+func NewDatabaseObjectWithLabels(name string, labels map[string]string, spec *dbobjectv1.DatabaseObjectSpec) (*dbobjectv1.DatabaseObject, error) {
+	databaseObject := &dbobjectv1.DatabaseObject{
+		Kind:    types.KindDatabaseObject,
+		Version: types.V1,
+		Metadata: &headerv1.Metadata{
+			Name:      name,
+			Namespace: defaults.Namespace,
+			Labels:    labels,
+		},
+		Spec: spec,
+	}
+
+	err := ValidateDatabaseObject(databaseObject)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return databaseObject, nil
+}
+
+// ValidateDatabaseObject checks if *dbobjectv1.DatabaseObject is valid.
+func ValidateDatabaseObject(obj *dbobjectv1.DatabaseObject) error {
+	if obj == nil {
+		return trace.BadParameter("object must be non-nil")
```

Comment:
```suggestion
		return trace.BadParameter("database object must be non-nil")
```

Just "object" could be about anything, which can make this error confusing if it's coming from several layers down.

Similarly for other errors.

Reply:
Isn't this the reason for using `trace` package in the first place? You are getting a full stack trace information, which gives you this information and more.

Reply:
I've changed the message to indicate the object type throwing the error.